src/state/error.tsx

```diff
@@ -0,0 +1,23 @@
+import { createContext, PropsWithChildren, useState } from 'react'
+
+export const ErrorContext = createContext<{
+  error: Error | undefined
+  setError: (e: Error) => void
+}>({
+  error: undefined,
+  setError: () => null,
+})
+
+export const ErrorContextProvider = (props: PropsWithChildren<Record<string, unknown>>) => {
```

Comment:
Just call this `Provider`, and then alias it when importing (`import { Provider as ErrorProvider } from 'state/error'`).